src/import-xml/import-xml.service.ts

```diff
@@ -106,6 +110,55 @@ export class ImportXmlService {
     };
   }
 
+  async reprocessXML(id: string) {
+    const queryRunner = AppDataSource.createQueryRunner();
+    const importXml = await AppDataSource.createQueryBuilder()
+      .select('i')
+      .from(ImportXml, 'i')
+      .where('i.professor_name=:id', { id: id })
```

Comment:
The id is used to search for professor_name? Probably should be another field here, not the professor_name

Reply:
The id is actually the professor name, the query is used to get the most recent import of the professor by its name, because the id is generated randomly, i fixed it.